packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -391,6 +392,12 @@ export class OdspDocumentService implements IDocumentService {
         nonAfdUrl: string,
         afdUrl?: string,
     ): Promise<IDocumentDeltaConnection> {
+        let clientToUse = client;
+        if (this.hostPolicy.sessionOptions?.unauthenticatedUserDisplayName) {
+            clientToUse = cloneDeep(client);
+            (clientToUse.user as IUnauthenticatedUser).displayName =
```

Comment:
the server will clobber this. any changes to the user will not be round tripped

Reply:
https://github.com/microsoft/FluidFramework/blob/62e22d418c784bfbc97cbfcd3505cf0f6d5e5e89/server/routerlicious/packages/lambdas/src/alfred/index.ts#L237

Reply:
Push needs to do work to handle "on behalf of" mode and to not use claims in authorization token. This was proposed by ODSP/Push folks.

Reply:
I'd rather stick with the existing loader options until we have a whole picture, as i still don't see how this can be done securely

Reply:
Alfred will clobber this. But @GaryWilber has a different alfred so he can pass this through. I also don't understand how to secure this flow.

Reply:
After discussing with ODSP we decided on alternative approach. Instead of passing display name as part of joinSession call which will use POST body for that. Please take a look and see if you have any concerns with this.

Passing user name is spoofable by design but will be used only when proof token flow is in use. This will be checked on backend.